app.py

```diff
@@ -1,72 +1,64 @@
 #!/usr/bin/env python3
 
-import json
 import os
-import cherrypy
-from elasticsearch import Elasticsearch
-from lib.model.part import Part
-from lib.render import PHRender
+from pathlib import Path
 
+from elasticsearch import AsyncElasticsearch
+from quart import Quart, redirect, request, render_template, safe_join, send_file
 
-class Search:
-    def __init__(self):
-        self.phrender = PHRender()
+from lib.model.part import Part
 
-        self.env = self.phrender.env
-        self.template = self.phrender.get_class_template(self.__class__)
 
-        if 'ELASTICSEARCH' in os.environ:
-            self.es = Elasticsearch([os.environ['ELASTICSEARCH']])
-        else:
-            self.es = Elasticsearch()
+app = Quart(__name__)
 
-    def render(self, page=None):
-        """Render the template associated with the page.
+STATIC_DIR = Path(__file__).parent / '_site'
 
-        If the client requests JSON, then this returns a JSON dump of the
-        """
+@app.route("/", defaults={'path': ""})
+@app.route("/<path:path>")
+async def serve(path: str):
+    p = safe_join(STATIC_DIR, path)
 
-        if page is None:
-            page = {}
+    if p.is_dir():
+        p /= 'index.html'
 
-        if self._accepts_json() or (not self._explicitly_accepts_html()):
-            cherrypy.response.headers['Content-Type'] = 'application/json; charset=utf-8'
-            return json.dumps(page, indent=2, sort_keys=True).encode('utf-8')
+    if p.is_file():
+        return await send_file(p)
 
-        return self.template.render(page=page)
+    return "File not found", 404
 
-    @staticmethod
-    def _accepts(mimetype):
-        headers = cherrypy.request.headers
-        # The .replace() isn't technically correct.
-        accepted = headers.get('Accept', '').replace(';', ',').split(',')
-        return mimetype in accepted
 
-    def _explicitly_accepts_html(self):
-        return self._accepts('text/html')
 
-    def _accepts_json(self):
-        return self._accepts('application/json')
+if 'ELASTICSEARCH' in os.environ:
+    es = AsyncElasticsearch([os.environ['ELASTICSEARCH']])
```

Comment:
I feel a bit dirty, creating those resources (`app`, `es`, ...) at import-time.
ASGI servers support calling a function to get the app object, I should do that.